frontend/src/metabase/public/containers/PublicAction/PublicActionLoader.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback, useState } from "react";
+import { connect } from "react-redux";
+
+import { useOnMount } from "metabase/hooks/use-on-mount";
+import { useSafeAsyncFunction } from "metabase/hooks/use-safe-async-function";
+import { setErrorPage } from "metabase/redux/app";
+import { PublicApi } from "metabase/services";
+
+import type { WritebackAction } from "metabase-types/api";
+import type { AppErrorDescriptor } from "metabase-types/store";
+
+import EmbedFrame from "../../components/EmbedFrame";
+import PublicAction from "./PublicAction";
+import {
+  LoadingAndErrorWrapper,
+  ContentContainer,
+} from "./PublicAction.styled";
+
+interface OwnProps {
+  params: { uuid: string };
+}
+
+interface DispatchProps {
+  setErrorPage: (error: AppErrorDescriptor) => void;
+}
+
+type Props = OwnProps & DispatchProps;
+
+const mapDispatchToProps = {
+  setErrorPage,
+};
+
+function PublicActionLoader({ params, setErrorPage }: Props) {
+  const [action, setAction] = useState<WritebackAction | null>(null);
+  const fetchAction = useSafeAsyncFunction(PublicApi.action);
+
+  useOnMount(() => {
+    async function loadAction() {
+      try {
+        const action = await fetchAction({ uuid: params.uuid });
+        setAction(action);
+      } catch (error) {
+        setErrorPage(error as AppErrorDescriptor);
+      }
+    }
+    loadAction();
```

Comment:
I like that you name the function instead of just writing its body. This helps convey what you're trying to do, however, what do you think about naming the function passed to `useOnMount` instead?

e.g.

```ts
  useOnMount(async function loadAction() {
    try {
      const action = await fetchAction({ uuid: params.uuid });
      setAction(action);
    } catch (error) {
      setErrorPage(error as AppErrorDescriptor);
    }
  });
```

Reply:
I'm not sure if we do it anywhere else? I gave it a name primarily because `useEffect` function can't be async

Reply:
That makes sense, I didn't realize we can't use async with `useEffect`. I also couldn't find we use this anywhere either. But sometimes, it's a good idea to give the anonymous function a name still.